### ISSUE TITLE
Passing `nil` as trait argument raises an error from v5

### ISSUE THREAD

**Issue by mtsmfm** [thoughtbot/factory_bot#1267]
I've noticed `FactoryBot.create(:factory, nil)` raises an error from v5.

```ruby
require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'
  if ENV['FACTORY_BOT_4']
    gem 'factory_bot', '4.11.1'
  else
    gem 'factory_bot', '5.0.1'
  end
end


class X
end

FactoryBot.define do
  factory :x do
  end
end

FactoryBot.build(:x, nil)
puts 'ok'

# $ ruby xxx.rb
# Traceback (most recent call last):
# 4: from xxx.rb:21:in `<main>'
# 3: from /usr/local/bundle/gems/factory_bot-5.0.1/lib/factory_bot/strategy_syntax_method_registrar.rb:20:in `block in define_singular_strategy_method'
# 2: from /usr/local/bundle/gems/factory_bot-5.0.1/lib/factory_bot/strategy_syntax_method_registrar.rb:20:in `new'
# 1: from /usr/local/bundle/gems/factory_bot-5.0.1/lib/factory_bot/factory_runner.rb:8:in `initialize'
# /usr/local/bundle/gems/factory_bot-5.0.1/lib/factory_bot/factory_runner.rb:8:in `map': undefined method `to_sym' for nil:NilClass (NoMethodError)


# $ FACTORY_BOT_4=true ruby xxx.rb
# ok
```

Is this intentional?

**Comment by composerinteralia** [comment]
I can't say I know what the expected behavior is for passing `nil` there, but that particular error is definitely not expected. This came up also in https://github.com/thoughtbot/factory_bot/issues/1259. Thanks!

**Comment by composerinteralia** [comment]
I just released 5.0.2 which has a fix for this
